staging/src/k8s.io/apiserver/pkg/features/kube_features.go

```diff
@@ -232,6 +232,18 @@ const (
 	//
 	// Allow the API server to serve consistent lists from cache
 	ConsistentListFromCache featuregate.Feature = "ConsistentListFromCache"
+
+	// owner: @saschagrunert
+	// alpha: v1.28
+	//
+	// Enables user namespace support for Pod Security Standards. Enabling this
+	// feature will modify all Pod Security Standard rules to allow setting:
+	// spec[.*].securityContext.[runAsNonRoot,runAsUser,allowPrivilegeEscalation,capabilities]
+	// This feature gate should only be enabled if all nodes in the cluster
+	// support the user namespace feature and have it enabled. The feature gate
+	// will not graduate or be enabled by default in future Kubernetes
+	// releases.
+	UserNamespacesPodSecurityStandards featuregate.Feature = "UserNamespacesPodSecurityStandards"
```

Comment:
Oh, this feature gate doesn't belong in generic k8s.io/apiserver features, it's specific to kubernetes components. Move this to pkg/features/kube_features.go

That's going to make plumbing difficult :-/

I'm remembering more about this now... we explicitly didn't want the meaning of PSS policies to be stateful or vary between clusters or rely on feature toggles.

A hacky way to make the kube-apiserver-constructed admission instance be informed by kube-apiserver-specific gates is to do something like this:

In staging/src/k8s.io/pod-security-admission/policy/helpers.go:
```go
var relaxPolicyForUserNamespacePods = &atomic.Bool{}

// RelaxPolicyForUserNamespacePods allows opting into relaxing runAsUser / runAsNonRoot restricted policies for user namespace pods,
// before the usernamespace feature has reached GA and propagated to the oldest supported nodes.
// This should only be opted into in clusters where the administrator ensures all nodes in the cluster enable the user namespace feature.
func RelaxPolicyForUserNamespacePods(relax bool) {
  relaxPolicyForUserNamespacePods.Store(relax)
} 

func relaxPolicyForUserNamespacePod(podSpec *corev1.PodSpec) bool {
	return relaxPolicyForUserNamespacePods.Load() && podSpec != nil && podSpec.HostUsers != nil && !*podSpec.HostUsers
}
```

In plugin/pkg/admission/security/podsecurity.Plugin#InspectFeatureGates:
```go
policy.RelaxPolicyForUserNamespacePods.Store(features.UserNamespacesPodSecurityStandards)
```

Reply:
Thank you for the pointers, I'll change it that way.